diff --git a/node_modules/@react-native-community/hooks/lib/useAppState.js b/node_modules/@react-native-community/hooks/lib/useAppState.js
index eebe6a5..3fc2124 100644
--- a/node_modules/@react-native-community/hooks/lib/useAppState.js
+++ b/node_modules/@react-native-community/hooks/lib/useAppState.js
@@ -10,9 +10,9 @@ function useAppState() {
         setAppState(newState);
     }
     react_1.useEffect(function () {
-        react_native_1.AppState.addEventListener('change', onChange);
+        var sub = react_native_1.AppState.addEventListener('change', onChange);
         return function () {
-            react_native_1.AppState.removeEventListener('change', onChange);
+            sub.remove();
         };
     }, []);
     return appState;
