diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index d55c34b..4263127 100755
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -2394,6 +2394,8 @@ export interface ViewProps
      */
     hitSlop?: Insets | undefined;
 
+    opacity?: number | undefined;
+
     /**
      * Invoked on mount and layout changes with
      *
