diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index e9cced2..d8ed279 100755
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -2566,6 +2566,9 @@ export interface ViewProps
         PointerEvents,
         AccessibilityProps {
     children?: React.ReactNode;
+
+    opacity?: number;
+
     /**
      * This defines how far a touch event can start away from the view.
      * Typical interface guidelines recommend touch targets that are at least
