diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index 0e0a4b5..2401bd3 100755
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -2487,6 +2487,8 @@ export interface ViewProps
      */
     hitSlop?: Insets | undefined;
 
+    opacity?: number;
+
     /**
      * Invoked on mount and layout changes with
      *
