diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index f05993c..652b28b 100755
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -2506,6 +2506,8 @@ export interface ViewProps
      */
     hitSlop?: Insets;
 
+    opacity?: number;
+
     /**
      * Invoked on mount and layout changes with
      *
