diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index b301eee..65b54a6 100755
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -2386,6 +2386,8 @@ export interface ViewProps
         Touchable,
         AccessibilityProps {
     children?: React.ReactNode;
+    opacity?: number;
+
     /**
      * This defines how far a touch event can start away from the view.
      * Typical interface guidelines recommend touch targets that are at least
