diff --git a/node_modules/react-native-share/android/build.gradle b/node_modules/react-native-share/android/build.gradle
index 562709f..09651bc 100644
--- a/node_modules/react-native-share/android/build.gradle
+++ b/node_modules/react-native-share/android/build.gradle
@@ -20,11 +20,11 @@ buildscript {
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion safeExtGet('compileSdkVersion', 28)
-    buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
+    compileSdkVersion safeExtGet('compileSdkVersion', 29)
+    buildToolsVersion safeExtGet('buildToolsVersion', '29.0.3')
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', 16)
-        targetSdkVersion safeExtGet('targetSdkVersion', 28)
+        targetSdkVersion safeExtGet('targetSdkVersion', 29)
     }
     lintOptions {
         abortOnError false
diff --git a/node_modules/react-native-share/windows/.npmignore b/node_modules/react-native-share/windows/.gitignore
similarity index 100%
rename from node_modules/react-native-share/windows/.npmignore
rename to node_modules/react-native-share/windows/.gitignore
