diff --git a/node_modules/react-native-markdown-view/index.d.ts b/node_modules/react-native-markdown-view/index.d.ts
index 9b2ad53..d24f405 100644
--- a/node_modules/react-native-markdown-view/index.d.ts
+++ b/node_modules/react-native-markdown-view/index.d.ts
@@ -172,6 +172,8 @@ declare module 'react-native-markdown-view' {
      * string (first and only argument).
      */
     onLinkPress?: (url: string) => void;
+
+    children: string;
   }

   /**
