diff --git a/node_modules/react-native-keyevent/ios/RNKeyEvent.m b/node_modules/react-native-keyevent/ios/RNKeyEvent.m
index 8c6eaea..779cc00 100644
--- a/node_modules/react-native-keyevent/ios/RNKeyEvent.m
+++ b/node_modules/react-native-keyevent/ios/RNKeyEvent.m
@@ -32,7 +32,7 @@ RCT_EXPORT_MODULE();
 }
 
 - (NSString *)getKeys {
-    return  [NSString stringWithFormat:@"!,1,2,3,4,5,6,7,8,9,0,a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,A,B,C,D,E,F,G,H,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y,Z,~,-,\r,%@,%@,%@,%@",UIKeyInputLeftArrow, UIKeyInputRightArrow, UIKeyInputUpArrow, UIKeyInputDownArrow];
+    return  [NSString stringWithFormat:@"!,1,2,3,4,5,6,7,8,9,0,a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,A,B,C,D,E,F,G,H,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y,Z,~,-, ,\r,%@,%@,%@,%@",UIKeyInputLeftArrow, UIKeyInputRightArrow, UIKeyInputUpArrow, UIKeyInputDownArrow];
 }
 
 - (void)sendKeyEvent:(NSString *)keyString {
