diff --git a/node_modules/react-native-markdown-view/renders.js b/node_modules/react-native-markdown-view/renders.js
index 490e62b..f46e35b 100644
--- a/node_modules/react-native-markdown-view/renders.js
+++ b/node_modules/react-native-markdown-view/renders.js
@@ -82,7 +82,7 @@ function paragraphRenderer() {
 
 function textContentRenderer(styleName, styleName2) {
   return (node: InlineContentNode, output: OutputFunction, state: RenderState, styles: RenderStyles) => (
-    <Text key={state.key} style={styleName2 ? [styles[styleName], styles[styleName2]] : styles[styleName]}>
+    <Text selectable key={state.key} style={styleName2 ? [styles[styleName], styles[styleName2]] : styles[styleName]}>
       {typeof node.content === 'string' ? node.content : output(node.content, state)}
     </Text>
   )
