diff --git a/node_modules/react-native/Libraries/Components/View/ViewPropTypes.d.ts b/node_modules/react-native/Libraries/Components/View/ViewPropTypes.d.ts
index 20ff434..a735cc1 100644
--- a/node_modules/react-native/Libraries/Components/View/ViewPropTypes.d.ts
+++ b/node_modules/react-native/Libraries/Components/View/ViewPropTypes.d.ts
@@ -181,6 +181,8 @@ export interface ViewProps
    */
   hitSlop?: null | Insets | number | undefined;

+  opacity?: number;
+
   /**
    * Used to reference react managed views from native code.
    */
