diff --git a/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m b/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m
index 0245c31..7760675 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m
@@ -34,9 +34,10 @@ -(instancetype)init:(NSString*)buttonId withCustomView:(RCTRootView *)reactView
 	reactView.sizeFlexibility = RCTRootViewSizeFlexibilityWidthAndHeight;
 	reactView.delegate = self;
 	reactView.backgroundColor = [UIColor clearColor];
-    reactView.hidden = CGRectEqualToRect(reactView.frame, CGRectZero);
+  reactView.hidden = CGRectEqualToRect(reactView.frame, CGRectZero);
+  reactView.translatesAutoresizingMaskIntoConstraints = NO;
     
-	[NSLayoutConstraint deactivateConstraints:reactView.constraints];
+  [NSLayoutConstraint deactivateConstraints:reactView.constraints];
 	self.widthConstraint = [NSLayoutConstraint constraintWithItem:reactView
 														attribute:NSLayoutAttributeWidth
 														relatedBy:NSLayoutRelationEqual
