diff --git a/node_modules/react-native-settings-components/src/picker/picker.jsx b/node_modules/react-native-settings-components/src/picker/picker.jsx
index a599dde..f29ea55 100644
--- a/node_modules/react-native-settings-components/src/picker/picker.jsx
+++ b/node_modules/react-native-settings-components/src/picker/picker.jsx
@@ -125,7 +125,7 @@ class SettingsPicker extends Component {
     });
   }
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
     this.setState({
       pickerValue: nextProps.multi && isArray(nextProps.value)
         ? nextProps.value : [nextProps.value],
