diff --git a/node_modules/react-native-svg/src/index.d.ts b/node_modules/react-native-svg/src/index.d.ts
index 2488ff3..9e60293 100644
--- a/node_modules/react-native-svg/src/index.d.ts
+++ b/node_modules/react-native-svg/src/index.d.ts
@@ -332,6 +332,7 @@ export interface PatternProps {
   patternContentUnits?: Units;
   viewBox?: string;
   preserveAspectRatio?: string;
+  children?: React.ReactNode;
 }
 export const Pattern: React.ComponentClass<PatternProps>;
 export type Pattern = React.ComponentClass<PatternProps>;
@@ -392,6 +393,8 @@ export interface SvgProps extends GProps, ReactNative.ViewProperties {
   preserveAspectRatio?: string;
   color?: Color;
   title?: string;
+  faction?: string;
+  border?: string;
 }
 
 // Svg is both regular and default exported
@@ -437,6 +440,7 @@ export interface TextProps extends TextSpecificProps {
   rotate?: NumberArray;
   opacity?: NumberProp;
   inlineSize?: NumberProp;
+  children?: string;
 }
 export const Text: React.ComponentClass<TextProps>;
 export type Text = React.ComponentClass<TextProps>;
