diff --git a/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m b/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m
index ed152c4..2388f72 100644
--- a/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m
+++ b/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m
@@ -83,7 +83,6 @@ - (void)updateBackgroundAppearance {
         self.navigationController.navigationBar.translucent = YES;
     } else {
         self.navigationController.navigationBar.translucent = NO;
-        self.navigationController.navigationBar.barTintColor = nil;
     }
 }

