diff --git a/node_modules/react-native-pan-pinch-view/src/index.tsx b/node_modules/react-native-pan-pinch-view/src/index.tsx
index c2f04e4..d1a7fe8 100644
--- a/node_modules/react-native-pan-pinch-view/src/index.tsx
+++ b/node_modules/react-native-pan-pinch-view/src/index.tsx
@@ -1,4 +1,4 @@
-import React, { forwardRef, useImperativeHandle } from 'react';
+import React, { forwardRef, useEffect, useImperativeHandle } from 'react';
 import { Platform, StyleSheet, View } from 'react-native';
 import {
   Gesture,
@@ -29,8 +29,10 @@ export default forwardRef(function PanPinchView(
     minScale = 0.5,
     maxScale = 4,
     initialScale = 1,
+    initialOffset = { x: 0, y: 0 },
     onTranslationFinished = undefined,
     children,
+    backgroundStyle,
   }: PanPinchViewProps,
   ref: React.Ref<PanPinchViewRef>
 ) {
@@ -42,7 +44,7 @@ export default forwardRef(function PanPinchView(
 
   const translation = useVector(0, 0);
   const adjustedFocal = useVector(0, 0);
-  const offset = useVector(0, 0);
+  const offset = useVector(initialOffset.x, initialOffset.y);
   const origin = useVector(0, 0);
 
   const isPinching = useSharedValue(false);
@@ -321,6 +323,7 @@ export default forwardRef(function PanPinchView(
               width: containerDimensions.width,
               height: containerDimensions.height,
             },
+            backgroundStyle,
           ]}
         >
           <Animated.View
diff --git a/node_modules/react-native-pan-pinch-view/src/types.d.ts b/node_modules/react-native-pan-pinch-view/src/types.d.ts
index 88d57e9..a946f83 100644
--- a/node_modules/react-native-pan-pinch-view/src/types.d.ts
+++ b/node_modules/react-native-pan-pinch-view/src/types.d.ts
@@ -1,4 +1,5 @@
 import React from 'react';
+import { ViewStyle } from 'react-native';
 
 type Dimensions = {
   width: number;
@@ -38,6 +39,13 @@ export type PanPinchViewProps = {
    */
   initialScale?: number;
 
+  initialOffset?: {
+    x: number;
+    y: number;
+  };
+
+  backgroundStyle?: ViewStyle;
+
   /**
    * Callback function called when translation is finished updating.
    */
