diff --git a/node_modules/react-native/index.js b/node_modules/react-native/index.js
index d59ba34..8e4e926 100644
--- a/node_modules/react-native/index.js
+++ b/node_modules/react-native/index.js
@@ -440,6 +440,7 @@ module.exports = {
       'ColorPropType has been removed from React Native. Migrate to ' +
         "ColorPropType exported from 'deprecated-react-native-prop-types'.",
     );
+    return require('deprecated-react-native-prop-types').ColorPropType;
   },
   get EdgeInsetsPropType(): $FlowFixMe {
     invariant(
@@ -447,6 +448,7 @@ module.exports = {
       'EdgeInsetsPropType has been removed from React Native. Migrate to ' +
         "EdgeInsetsPropType exported from 'deprecated-react-native-prop-types'.",
     );
+    return require('deprecated-react-native-prop-types').EdgeInsetsPropType;
   },
   get PointPropType(): $FlowFixMe {
     invariant(
@@ -454,6 +456,7 @@ module.exports = {
       'PointPropType has been removed from React Native. Migrate to ' +
         "PointPropType exported from 'deprecated-react-native-prop-types'.",
     );
+    return require('deprecated-react-native-prop-types').PointPropType;
   },
   get ViewPropTypes(): $FlowFixMe {
     invariant(
@@ -461,6 +464,7 @@ module.exports = {
       'ViewPropTypes has been removed from React Native. Migrate to ' +
         "ViewPropTypes exported from 'deprecated-react-native-prop-types'.",
     );
+    return require('deprecated-react-native-prop-types').ViewPropTypes;
   },
 };
 
