diff --git a/node_modules/react-native-pan-pinch-view/src/index.tsx b/node_modules/react-native-pan-pinch-view/src/index.tsx
index c2f04e4..978b3fc 100644
--- a/node_modules/react-native-pan-pinch-view/src/index.tsx
+++ b/node_modules/react-native-pan-pinch-view/src/index.tsx
@@ -31,6 +31,7 @@ export default forwardRef(function PanPinchView(
     initialScale = 1,
     onTranslationFinished = undefined,
     children,
+    backgroundStyle,
   }: PanPinchViewProps,
   ref: React.Ref<PanPinchViewRef>
 ) {
@@ -321,6 +322,7 @@ export default forwardRef(function PanPinchView(
               width: containerDimensions.width,
               height: containerDimensions.height,
             },
+            backgroundStyle,
           ]}
         >
           <Animated.View
diff --git a/node_modules/react-native-pan-pinch-view/src/types.d.ts b/node_modules/react-native-pan-pinch-view/src/types.d.ts
index 88d57e9..5e4976e 100644
--- a/node_modules/react-native-pan-pinch-view/src/types.d.ts
+++ b/node_modules/react-native-pan-pinch-view/src/types.d.ts
@@ -1,4 +1,5 @@
 import React from 'react';
+import { ViewStyle } from 'react-native';
 
 type Dimensions = {
   width: number;
@@ -38,6 +39,8 @@ export type PanPinchViewProps = {
    */
   initialScale?: number;
 
+  backgroundStyle?: ViewStyle;
+
   /**
    * Callback function called when translation is finished updating.
    */
