diff --git a/node_modules/react-native-side-menu/index.js b/node_modules/react-native-side-menu/index.js
index ed51bdc..c1084f7 100644
--- a/node_modules/react-native-side-menu/index.js
+++ b/node_modules/react-native-side-menu/index.js
@@ -103,7 +103,7 @@ export default class SideMenu extends React.Component {
     this.state.left.addListener(({value}) => this.props.onSliding(Math.abs((value - this.state.hiddenMenuOffset) / (this.state.openMenuOffset - this.state.hiddenMenuOffset))));
   }
 
-  componentWillMount(): void {
+  UNSAFE_componentWillMount(): void {
     this.responder = PanResponder.create({
       onStartShouldSetResponderCapture: this.onStartShouldSetResponderCapture,
       onMoveShouldSetPanResponder: this.onMoveShouldSetPanResponder,
@@ -113,7 +113,7 @@ export default class SideMenu extends React.Component {
     });
   }
 
-  componentWillReceiveProps(props: Props): void {
+  UNSAFE_componentWillReceiveProps(props: Props): void {
     if (typeof props.isOpen !== 'undefined' && this.isOpen !== props.isOpen && (props.autoClosing || this.isOpen === false)) {
       this.openMenu(props.isOpen);
     }
