diff --git a/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m b/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m
index 20af99d..a9ee016 100644
--- a/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m
+++ b/node_modules/react-native-navigation/lib/ios/TopBarPresenter.m
@@ -89,7 +89,6 @@
         self.navigationController.navigationBar.translucent = YES;
     } else {
         self.navigationController.navigationBar.translucent = NO;
-        self.navigationController.navigationBar.barTintColor = nil;
     }
 }
 
