diff --git a/node_modules/react-native-spring-scrollview/SpringScrollView.js b/node_modules/react-native-spring-scrollview/SpringScrollView.js
index 7d47fe7..a49e866 100644
--- a/node_modules/react-native-spring-scrollview/SpringScrollView.js
+++ b/node_modules/react-native-spring-scrollview/SpringScrollView.js
@@ -249,13 +249,14 @@ export class SpringScrollView extends React.PureComponent<SpringScrollViewPropTy
   }

   beginRefresh() {
-    if (!this.props.loadingFooter || this.props.loadingFooter.height <= 0)
+    if (!this.props.refreshHeader || this.props.refreshHeader.height <= 0)
       return Promise.reject(
-        "SpringScrollView: call beginRefresh without loadingFooter or loadingFooter height"
+        "SpringScrollView: call beginRefresh without refreshHeader or refreshHeader height"
       );
+    this._toRefreshStatus('refreshing');
     return this.scrollTo({
       x: this._contentOffset.x,
-      y: -this.props.loadingFooter.height - 1,
+      y: -this.props.refreshHeader.height - 1,
     });
   }

diff --git a/node_modules/react-native-spring-scrollview/SpringScrollViewNative.js b/node_modules/react-native-spring-scrollview/SpringScrollViewNative.js
index b81cd8c..e7ad5fd 100644
--- a/node_modules/react-native-spring-scrollview/SpringScrollViewNative.js
+++ b/node_modules/react-native-spring-scrollview/SpringScrollViewNative.js
@@ -64,6 +64,7 @@ export class SpringScrollViewNativeAdapter extends React.Component {
         onCustomScrollEndDrag={(e) =>
           this.props.onScrollEndDrag && this.props.onScrollEndDrag(e)
         }
+	onMoveShouldSetResponderCapture={() => { return true; }}
       />
     );
   }
