diff --git a/node_modules/react-native-navigation/lib/android/app/build/generated/res/pngs/reactNative60/debug/drawable-anydpi-v21/ic_arrow_back_black_24dp.xml b/node_modules/react-native-navigation/lib/android/app/build/generated/res/pngs/reactNative60/debug/drawable-anydpi-v21/ic_arrow_back_black_24dp.xml
ios/RNNUIBarButtonItem.m b/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m
index 32b7810..8489558 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNUIBarButtonItem.m
@@ -37,6 +37,7 @@ -(instancetype)init:(NSString*)buttonId withCustomView:(RCTRootView *)reactView
 	reactView.delegate = self;
 	reactView.backgroundColor = [UIColor clearColor];
 	reactView.hidden = YES;
+	reactView.translatesAutoresizingMaskIntoConstraints = NO;
 
 	self.widthConstraint = [NSLayoutConstraint constraintWithItem:reactView
 														attribute:NSLayoutAttributeWidth
