diff --git a/node_modules/react-native-navigation/ReactNativeNavigation.podspec b/node_modules/react-native-navigation/ReactNativeNavigation.podspec
index 1777dc6..1f3438e 100644
--- a/node_modules/react-native-navigation/ReactNativeNavigation.podspec
+++ b/node_modules/react-native-navigation/ReactNativeNavigation.podspec
@@ -15,7 +15,7 @@ Find.find(start_dir) do |path|
   end
 end

-swift_project = swift_delegate_path && File.exist?(swift_delegate_path)
+swift_project = false; # swift_delegate_path && File.exist?(swift_delegate_path)

 # Debug output
 if swift_project
@@ -51,9 +51,38 @@ Pod::Spec.new do |s|

   folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32 -DFOLLY_CFG_NO_COROUTINES=1'

+  header_search_paths = [
+    "$(PODS_ROOT)/boost",
+    "$(PODS_ROOT)/boost-for-react-native",
+    "$(PODS_ROOT)/RCT-Folly",
+    "$(PODS_ROOT)/Headers/Private/React-Core",
+    "$(PODS_ROOT)/Headers/Private/Yoga",
+  ]
+
+  if fabric_enabled && ENV['USE_FRAMEWORKS']
+    header_search_paths.concat([
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-FabricComponents/React_FabricComponents.framework/Headers/react/renderer/textlayoutmanager/platform/ios"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-FabricComponents/React_FabricComponents.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-nativeconfig/React_nativeconfig.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-RuntimeApple/React_RuntimeApple.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-RuntimeCore/React_RuntimeCore.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-performancetimeline/React_performancetimeline.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-runtimescheduler/React_runtimescheduler.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-rendererconsistency/React_rendererconsistency.framework/Headers"',
+      '"$(PODS_ROOT)/Headers/Public/ReactCommon"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-jserrorhandler/React_jserrorhandler.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/RCT-Folly/folly.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/fmt/fmt.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-utils/React_utils.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-debug/React_debug.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-rendererdebug/React_rendererdebug.framework/Headers"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-jsinspector/jsinspector_modern.framework/Headers"',
+    ])
+  end
+
   # Base xcconfig settings
   xcconfig_settings = {
-      'HEADER_SEARCH_PATHS' => '"$(PODS_ROOT)/boost" "$(PODS_ROOT)/boost-for-react-native"  "$(PODS_ROOT)/RCT-Folly" "$(PODS_ROOT)/Headers/Private/React-Core" "$(PODS_ROOT)/Headers/Private/Yoga"',
+      'HEADER_SEARCH_PATHS' => header_search_paths.join(' '),
       "CLANG_CXX_LANGUAGE_STANDARD" => "c++20",
       "OTHER_CPLUSPLUSFLAGS" => "-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1",
   }
diff --git a/node_modules/react-native-navigation/lib/ios/RNNEventEmitter.mm b/node_modules/react-native-navigation/lib/ios/RNNEventEmitter.mm
index fe03a35..c40f645 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNEventEmitter.mm
+++ b/node_modules/react-native-navigation/lib/ios/RNNEventEmitter.mm
@@ -1,7 +1,7 @@
 #import "RNNEventEmitter.h"
 #import "RNNUtils.h"
 #import "RNNTurboEventEmitter.h"
-#import <React-RuntimeApple/ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTHost.h>

 @implementation RNNEventEmitter { }

diff --git a/node_modules/react-native-navigation/lib/ios/RNNReactButtonView.h b/node_modules/react-native-navigation/lib/ios/RNNReactButtonView.h
index 6dd7d74..298a1de 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNReactButtonView.h
+++ b/node_modules/react-native-navigation/lib/ios/RNNReactButtonView.h
@@ -1,5 +1,5 @@
 #import "RNNComponentView.h"
-#import <React-RuntimeApple/ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTHost.h>
 #import <React/RCTSurfacePresenterStub.h>
 #import <React/RCTFabricSurface.h>
 #import <React/RCTSurfacePresenter.h>
diff --git a/node_modules/react-native-navigation/lib/ios/RNNReactView.mm b/node_modules/react-native-navigation/lib/ios/RNNReactView.mm
index 1d28787..4c66383 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNReactView.mm
+++ b/node_modules/react-native-navigation/lib/ios/RNNReactView.mm
@@ -3,7 +3,7 @@
 #import <React/RCTRootContentView.h>

 #ifdef RCT_NEW_ARCH_ENABLED
-#import <React-RuntimeApple/ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTHost.h>
 #import <React/RCTFabricSurface.h>
 #import <React/RCTSurfacePresenter.h>

diff --git a/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h b/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h
index 0c8f231..21ec6b1 100644
--- a/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h
+++ b/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h
@@ -4,7 +4,7 @@
 #import <UIKit/UIKit.h>

 #ifdef RCT_NEW_ARCH_ENABLED
-#import <React-RuntimeApple/ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTHost.h>
 #import "RNNTurboManager.h"
 #endif

diff --git a/node_modules/react-native-navigation/lib/ios/ScreenAnimationController.h b/node_modules/react-native-navigation/lib/ios/ScreenAnimationController.h
index 4d3c7f9..46c6f3f 100644
--- a/node_modules/react-native-navigation/lib/ios/ScreenAnimationController.h
+++ b/node_modules/react-native-navigation/lib/ios/ScreenAnimationController.h
@@ -5,7 +5,7 @@
 #import <React/RCTUIManagerUtils.h>

 #ifdef RCT_NEW_ARCH_ENABLED
-#import <React-RuntimeApple/ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTHost.h>
 #import <React/RCTSurfacePresenterStub.h>
 #endif

diff --git a/node_modules/react-native-navigation/lib/ios/TurboModules/RNNTurboManager.mm b/node_modules/react-native-navigation/lib/ios/TurboModules/RNNTurboManager.mm
index cd429d3..5e26af0 100644
--- a/node_modules/react-native-navigation/lib/ios/TurboModules/RNNTurboManager.mm
+++ b/node_modules/react-native-navigation/lib/ios/TurboModules/RNNTurboManager.mm
@@ -9,7 +9,7 @@
 #import "RNNReactComponentRegistry.h"
 #import "RNNReactRootViewCreator.h"
 #import "RNNTurboCommandsHandler.h"
-#import <React-RuntimeApple/ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTHost.h>

 @interface RNNTurboManager ()

