diff --git a/node_modules/react-native-track-player/index.d.ts b/node_modules/react-native-track-player/index.d.ts
index b3ea0be..212a719 100644
--- a/node_modules/react-native-track-player/index.d.ts
+++ b/node_modules/react-native-track-player/index.d.ts
@@ -165,6 +165,16 @@ declare namespace RNTrackPlayer {
     public getBufferedProgress: () => number;
   }
 
+  // Hooks
+
+  export function useTrackPlayerEvents(events: string[], handler: (event: string) => void): void;
+  export function useTrackPlayerProgress(interval?: number): {
+    position: number;
+    bufferedPosition: number;
+    duration: number;
+  };
+  export function usePlaybackState(): State;
+
   // Constants
 
   export const STATE_NONE: State;
